### PR TITLE
Fix snap server

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

Switching RPC to the minimal API by default broke snapshot fetch and vote account sanity checks

#### Summary of Changes

* restore snapshot fetch by moving `getGenesisHash` to the minimal API
* switch vote account sanity checks to opt-in

### CODE REVIEW

**Comment by t-nelson** [comment]
ok this works as advertised

took a little while to test 'cause apparently we ignore `--only-known-rpc` when `--no-fetch-snapshot` is in force :upside_down_face:

File: validator/src/main.rs [hunk 1]
```diff
@@ -543,8 +543,18 @@ pub fn main() {
...
                 .conflicts_with("no_voting")
                 .requires("entrypoint")
+                .hidden(true)
                 .help("Skip the RPC vote account sanity check")
         )
+        .arg(
+            Arg::with_name("check_vote_account")
+                .long("check-vote-account")
+                .takes_value(true)
+                .value_name("RPC_ENTRYPOINT")
```
Comment (mvines) [inline validator/src/main.rs:553]:
```suggestion
                .value_name("RPC_URL")
```

File: validator/src/main.rs [hunk 2]
```diff
@@ -543,8 +543,18 @@ pub fn main() {
...
+                .requires("entrypoint")
+                .conflicts_with_all(&["no_check_vote_account", "no_voting"])
+                .help("Sanity check vote account state at startup. The host at RPC_ENTRYPOINT must expose `--full-rpc-api`")
```
Comment (mvines) [inline validator/src/main.rs:556]:
```suggestion
                .help("Sanity check vote account state at startup. The JSON RPC endpoint at RPC_URL must expose `--full-rpc-api`")
```

File: validator/src/main.rs [hunk 3]
```diff
@@ -1913,10 +1923,15 @@ pub fn main() {
...
+    if matches.is_present("no_check_vote_account") {
+        info!("vote account sanity checks are no longer performed by default. --no-check-vote-account is deprecated and can be removed from the command line");
+    }
     let rpc_bootstrap_config = bootstrap::RpcBootstrapConfig {
         no_genesis_fetch: matches.is_present("no_genesis_fetch"),
         no_snapshot_fetch: matches.is_present("no_snapshot_fetch"),
-        no_check_vote_account: matches.is_present("no_check_vote_account"),
+        check_vote_account: matches
+            .value_of("check_vote_account")
+            .map(|v| v.to_string()),
```
Comment (mvines) [inline validator/src/main.rs:1934]:
```suggestion
            .map(|url| url.to_string()),
```

File: validator/src/bootstrap.rs [hunk 4]
```diff
@@ -603,7 +603,8 @@ mod without_incremental_snapshots {
             }
         })
         .map(|_| {
-            if !validator_config.voting_disabled && !bootstrap_config.no_check_vote_account {
+            if let Some(rpc_endpoint) = bootstrap_config.check_vote_account.as_ref() {
+                let rpc_client = RpcClient::new(rpc_endpoint);
```
Comment (mvines) [inline validator/src/bootstrap.rs:607]:
```suggestion
            if let Some(url) = bootstrap_config.check_vote_account.as_ref() {
                let rpc_client = RpcClient::new(url);
```

File: validator/src/bootstrap.rs [hunk 5]
```diff
@@ -943,7 +944,8 @@ mod with_incremental_snapshots {
                 )
             })
             .map(|_| {
-                if !validator_config.voting_disabled && !bootstrap_config.no_check_vote_account {
+                if let Some(rpc_endpoint) = bootstrap_config.check_vote_account.as_ref() {
+                    let rpc_client = RpcClient::new(rpc_endpoint);
```
Comment (mvines) [inline validator/src/bootstrap.rs:948]:
```suggestion
                if let Some(url) = bootstrap_config.check_vote_account.as_ref() {
                    let rpc_client = RpcClient::new(url);
```

**Review by mvines — DISMISSED** [review]
lgtm, just some bikesheding

**Review by CriesofCarrots — APPROVED** [review]
Wfm

**Pull request merged**